nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/shape/Diag.java

```diff
@@ -52,11 +51,19 @@ public Diag(@NonNull INDArray input, @NonNull INDArray output){
         super(null, new INDArray[]{input}, wrapOrNull(output));
     }
 
+    public Diag(SameDiff sameDiff, SDVariable input) {
+        this(sameDiff, new SDVariable[]{input}, false);
+    }
+
     public Diag(SameDiff sameDiff, SDVariable[] args, boolean inPlace) {
         super(null, sameDiff, args, inPlace);
 
     }
 
+    public Diag(SameDiff sameDiff, SDVariable input, boolean inPlace) {
```

Comment:
We shouldn't ever need to mix "inplace" with SameDiff, only applies to INDArray/ND4J case.
(I realize there was old bad constructor)
Let's get rid of that constructor arg and use inplace=false hardcoded

Reply:
Fixed.